artifactory/services/go/publishzipandmod.go

```
@@ -69,20 +97,27 @@ func addGoVersion(version string, urlPath *string) {
 // props - The properties to be assigned for each artifact
 // ext - The extension of the file: zip, mod, info. This extension will be joined with the version for the path. For example v1.2.3.info or v1.2.3.zip
 // urlPath - The url including the repository. For example: http://127.0.0.1/artifactory/api/go/go-local
-func (pwa *publishZipAndModApi) upload(localPath, moduleId, version, props, ext, urlPath string) error {
+func (pwa *publishZipAndModApi) upload(localPath, moduleId, version, props, ext, urlPath string) (*clientutils.FileTransferDetails, error) {
 	err := CreateUrlPath(moduleId, version, props, ext, &urlPath)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	addGoVersion(version, &urlPath)
 	details, err := fileutils.GetFileDetails(localPath)
 	if err != nil {
-		return err
+		return nil, err
 	}
 	utils.AddChecksumHeaders(pwa.clientDetails.Headers, details)
 	resp, _, err := pwa.client.UploadFile(localPath, urlPath, "", &pwa.clientDetails, GoUploadRetries, nil)
 	if err != nil {
-		return err
+		return nil, err
+	}
+	sha256 := resp.Header.Get("X-Checksum-Sha256")
+	if err != nil {
```

This error is not returned by the previos line.
Maybe it should be:
```
if sha256 == nil {
```